### ISSUE TITLE
Implementing stricter guidelines regarding the age of a project?

### ISSUE THREAD

**Issue by Rabenherz112** [awesome-selfhosted/awesome-selfhosted#3880]
Since a PR (https://github.com/awesome-selfhosted/awesome-selfhosted/pull/3822) might not be the best place for discussing potential changes to contribution guidelines, I created a separate Issue.

The suggestion revolves around enhancing the identification of "low-quality" projects and excluding them from the list by introducing a minimum lifespan requirement. This requirement would serve to filter out newly created projects that are more susceptible to bugs and may still be in the early development stages. Additionally, it would offer insights into the developers' dedication to maintaining and regularly updating their software.

In my opinion, it would be beneficial to implement a requirement of two months for the minimum lifespan of projects. This timeframe allows for a reasonable evaluation period to assess if developers are consistently working on their software. It strikes a balance between giving projects enough time to showcase their progress and preventing the exclusion of potentially awesome new software from the list.

What are your thought on this?

**Comment by Rabenherz112** [comment]
First of all, thank you nodiscc for your response and valuable insights on this topic.

> Any software project you are adding to the list was first released more than 6 months ago

While I highly respect your experience as the primary maintainer (please understand that I mean no disrespect), I believe that a 6-month timeframe is too strict. Although I agree that it aligns with our definition of unmaintained projects, it could exclude already excellent software from the list. For example, in the past few weeks, we have approved software such as #3849, #3857, #3921, #3798 that would not meet the new requirement. While it would only mean for these software, that they have to wait until they have matured enough (and pass the check that they are still actively maintained), it would still prevent them in the meantime from being discovered.

> [...] We could create a special label for such addition requests, and people interested in trying out new projects could look at these.

While I think adding an additional label to communicate with other contributors is a great idea to indicate which projects have not yet met the minimum lifespan requirement, I doubt that it would be discovered by "people interested in trying out new projects". In my experience, not many individuals who are not actively planning to contribute or add something to a project look at pull requests.

> So adding this guideline would not filter out all occurences of low-quality or unsuitable software.

Of course, if it seemed like I implied that implementing such a guideline would completely eliminate low-quality or unsuitable software through PRs, then I am sorry. I also acknowledge that it is impossible to entirely exclude such instances, regardless of how strict the addition requirements may be. But it would (hopefully) ensure a standardized level of quality and alleviate some of the workload from the reviewers.

Considering the points raised, would it be worth discussing and considering a slightly lower timeframe, maybe something like 4 months? This could maybe strike a balance between allowing newer but promising software to be included while still ensuring a certain level of maturity and active maintenance. It could potentially address the concern of excluding valuable projects and having the list not become too "static" without compromising the overall goal of maintaining quality standards.

Let me know what you think of this.

**Comment by Rabenherz112** [comment]
(Please note that the PRs I mentioned as examples were only selected for illustrative purposes. They may still be excluded even if the guidelines would be lower than 6 Months.)

**Comment by nodiscc** [comment]
I agree. I proposed this change in https://github.com/awesome-selfhosted/awesome-selfhosted/pull/3999
